GOALS.md

```diff
@@ -1,4 +1,4 @@
-Chapel Public Release Goals (version 1.15.0)
+Chapel Public Release Goals (version 1.16.0 pre-release)
```

Comment:
Is it time to just retire the GOALS.md file altogether?  I think of it as more of a "covering our butts" file from the earlier days of the release.  As we become more established and mature, I'd feel less exposed without it.  I/we can do this in a separate PR, though.